hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1382,6 +1384,10 @@ bool MDMParser::getSignalStrength(NetStatus &status)
     if (_init && _pwr) {
         MDM_INFO("\r\n[ Modem::getSignalStrength ] = = = = = = = = = =");
 
+        // Ensure modem is responsive
+        if (!_atOk())
+            goto cleanup;
+
```

Comment:
If the previous call had been unresponsive, then it would already be in cleanup.

```
        // Reformat the operator string to be numeric
        // (allows the capture of `mcc` and `mnc`)
        sendFormated("AT+COPS=3,2\r\n");
        if (RESP_OK != waitFinalResp(nullptr, nullptr, COPS_TIMEOUT))
            goto cleanup;
```

Based on the nature of the previous call, it seems safe to assume you have a responsive AT interface.

Reply:
Right!  I was confused by the logs I have here where COPS=3,2 is called many times in succession.  That's not continuing on to the COPS? call.